spock-core/src/main/java/org/spockframework/runtime/GlobalExtensionRegistry.java

```diff
@@ -76,6 +78,11 @@ public List<IGlobalExtension> getGlobalExtensions() {
     return globalExtensions;
   }
 
+  @Override
+  public <T extends IStatelessAnnotationDrivenExtension<?>> T getStatelessAnnotationDrivenExtension(Class<T> extensionClass) {
+    return extensionClass.cast(statelessExtensions.computeIfAbsent(extensionClass, this::instantiateAndConfigureExtension));
```

Comment:
I would expect, that the map must be synchronized or a `ConcurrentMap`.

Reply:
This only happens during initial discovery, which is single-threaded.